### PR TITLE
feat(gRPC): `ExecuteTransactions` and `SimulateTransactions` API

### PR DESCRIPTION
**Pull request by bingyanglin**
# Description of change

Converts the gRPC `TransactionExecutionService` from single-transaction RPCs (`ExecuteTransaction` / `SimulateTransaction`) to batch RPCs (`ExecuteTransactions` / `SimulateTransactions`). Each RPC now accepts a list of transactions and returns per-item results (success or error), so a single round-trip can submit multiple transactions while isolating failures to individual items.

- **Proto**: Replace `ExecuteTransactionRequest/Response` with `ExecuteTransactionsRequest/Response` wrapping `repeated ExecuteTransactionItem` and `repeated ExecuteTransactionResult` (oneof success/`google.rpc.Status` error). Same pattern for simulate (`SimulateTransactionsRequest/Response`, `SimulateTransactionItem`, `SimulateTransactionResult`, `SimulatedTransaction`). Remove `field_mask_transparent` from the execute response; the read mask now applies directly to `ExecutedTransaction` fields inside each result.
- **Server (`iota-grpc-server`)**:
  - Extract shared helpers: `validate_batch_size`, `parse_read_mask`, `parse_transaction_proto` to reduce duplication between execute and simulate paths
  - `execute_transactions` iterates items sequentially, calling `execute_single_transaction` per item and capturing per-item errors as `google.rpc.Status`
  - `simulate_transactions` follows the same pattern with `simulate_single_transaction`
  - Response type for simulate changes from `SimulateTransactionResponse` to `SimulatedTransaction` (the per-item payload)
- **Config (`iota-config`)**: Add `max_execute_transaction_batch_size` and `max_simulate_transaction_batch_size` to `GrpcApiConfig` (default 20 each)
- **Client (`iota-grpc-client`)**:
  - Add `execute_transactions` and `simulate_transactions` batch methods returning `Vec<Result<…>>`
  - Rewrite `execute_transaction` / `simulate_transaction` as thin wrappers delegating to the batch methods
  - Add `SimulateTransactionInput` struct for per-item simulation options
  - Implement `ProtoResult` for `ExecuteTransactionResult` and `SimulateTransactionResult`; add `Error::EmptyRequest` variant
- **Generated code**: Regenerated accessors, field info, and proto Rust bindings for all new/renamed messages
- **Codegen (`iota-proto-build`)**: Add `field_mask_util` generation support for `oneof` fields with `google.rpc.Status` error variants
- **E2E tests**: Update all execute/simulate tests to use batch APIs; add new tests for batch execution, partial failure, missing fields, empty request, and batch size limit enforcement

## Links to any relevant issues

None

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [x] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: crates/iota-e2e-tests/tests/grpc/v0/mod.rs [hunk 1]
```diff
@@ -55,16 +53,9 @@ impl_field_presence_checker!(ExecutedTransaction {
...
-    effects,
-    events,
-    checkpoint,
-    timestamp,
-    input_objects,
-    output_objects,
-});
+// ExecuteTransactionsResponse is no longer transparent — it holds
+// `repeated ExecuteTransactionResult`.  Tests should extract the
+// ExecutedTransaction from the first result and assert against that.
```
Comment (muXxer) [inline crates/iota-e2e-tests/tests/grpc/v0/mod.rs]:
```suggestion
```

File: crates/iota-grpc-types/src/read_masks.rs [hunk 2]
```diff
@@ -321,29 +321,29 @@ pub const OBJECT_REFERENCE: &str = "reference";
...
+// SimulatedTransaction — per-method field constants
 // ---------------------------------------------------------------------------
 
 /// Read mask for
-/// [`SimulateTransactionResponse::executed_transaction()`](crate::v0::transaction_execution_service::SimulateTransactionResponse::executed_transaction).
+/// [`SimulatedTransaction::executed_transaction()`](crate::v0::transaction_execution_service::SimulatedTransaction::executed_transaction).
 ///
 /// Includes all ExecutedTransaction sub-fields. To request specific
 /// sub-fields, use paths like `"executed_transaction.effects"`.
 pub const SIMULATE_RESPONSE_EXECUTED_TRANSACTION: &str = "executed_transaction";
```
Comment (muXxer) [inline crates/iota-grpc-types/src/read_masks.rs]: Should we rename those to `SIMULATED_TRANSACTION_EXECUTED_TRANSACTION` or something similar?

File: crates/iota-grpc-server/src/transaction_execution_service/mod.rs [hunk 3]
```diff
@@ -268,27 +347,19 @@ pub async fn execute_transaction(
...
         .collect::<Result<_, _>>()?;
 
-    // Create a source for the merge
     let source = TransactionReadSource {
         reader: reader.clone(),
         config,
         transaction: Some(sdk_transaction),
         signatures: Some(signatures),
         effects: Some(effects.effects),
         events,
```
Comment (muXxer) [inline crates/iota-grpc-server/src/transaction_execution_service/mod.rs:356]: Actually, do we wait for checkpoint inclusion?

**Review by muXxer — CHANGES_REQUESTED** [review]

**Review by muXxer — APPROVED** [review]
